src/data/matrix/basic.lean

```diff
@@ -364,108 +377,96 @@ theorem diagonal_mul_diagonal' [decidable_eq n] (d₁ d₂ : n → α) :
   (1 : matrix n n α).map f = 1 :=
 one_map f.map_zero f.map_one
 
 /-- A version of `map_zero` where `f` is a `zero_hom`. -/
-@[simp] lemma zero_hom_map_zero
-  {β : Type w} [has_zero β] (f : zero_hom α β) :
+@[simp] lemma zero_hom_map_zero [has_zero α] [has_zero β] (f : zero_hom α β) :
   (0 : matrix n n α).map f = 0 :=
 map_zero f.map_zero
 
 /-- A version of `map_zero` where `f` is a `add_monoid_hom`. -/
-@[simp] lemma add_monoid_hom_map_zero
-  {β : Type w} [add_monoid β] (f : α →+ β) :
+@[simp] lemma add_monoid_hom_map_zero [add_monoid α] [add_monoid β] (f : α →+ β) :
   (0 : matrix n n α).map f = 0 :=
 map_zero f.map_zero
 
 /-- A version of `map_zero` where `f` is a `add_equiv`. -/
-@[simp] lemma add_equiv_map_zero
-  {β : Type w} [add_monoid β] (f : α ≃+ β) :
+@[simp] lemma add_equiv_map_zero [add_monoid α] [add_monoid β] (f : α ≃+ β) :
   (0 : matrix n n α).map f = 0 :=
 map_zero f.map_zero
 
 /-- A version of `map_zero` where `f` is a `linear_map`. -/
-@[simp] lemma linear_map_map_zero {R : Type*} [semiring R]
-  {β : Type w} [add_comm_monoid β] [module R α] [module R β] (f : α →ₗ[R] β) :
+@[simp] lemma linear_map_map_zero [semiring R] [add_comm_monoid α] [add_comm_monoid β]
+  [module R α] [module R β] (f : α →ₗ[R] β) :
   (0 : matrix n n α).map f = 0 :=
 map_zero f.map_zero
 
 /-- A version of `map_zero` where `f` is a `linear_equiv`. -/
-@[simp] lemma linear_equiv_map_zero {R : Type*} [semiring R]
-  {β : Type w} [add_comm_monoid β] [module R α] [module R β] (f : α ≃ₗ[R] β) :
+@[simp] lemma linear_equiv_map_zero [semiring R] [add_comm_monoid α] [add_comm_monoid β]
+  [module R α] [module R β] (f : α ≃ₗ[R] β) :
   (0 : matrix n n α).map f = 0 :=
 map_zero f.map_zero
 
 /-- A version of `map_zero` where `f` is a `ring_hom`. -/
-@[simp] lemma ring_hom_map_zero
-  {β : Type w} [semiring β] (f : α →+* β) :
+@[simp] lemma ring_hom_map_zero [semiring α] [semiring β] (f : α →+* β) :
   (0 : matrix n n α).map f = 0 :=
 map_zero f.map_zero
 
 /-- A version of `map_zero` where `f` is a `ring_equiv`. -/
-@[simp] lemma ring_equiv_map_zero
-  {β : Type w} [semiring β] (f : α ≃+* β) :
+@[simp] lemma ring_equiv_map_zero [semiring α] [semiring β] (f : α ≃+* β) :
   (0 : matrix n n α).map f = 0 :=
 map_zero f.map_zero
 
-lemma is_add_monoid_hom_mul_left (M : matrix l m α) :
-  is_add_monoid_hom (λ x : matrix m n α, M ⬝ x) :=
-{ to_is_add_hom := ⟨matrix.mul_add _⟩, map_zero := matrix.mul_zero _ }
-
-lemma is_add_monoid_hom_mul_right (M : matrix m n α) :
-  is_add_monoid_hom (λ x : matrix l m α, x ⬝ M) :=
-{ to_is_add_hom := ⟨λ _ _, matrix.add_mul _ _ _⟩, map_zero := matrix.zero_mul _ }
-
-protected lemma sum_mul {β : Type*} (s : finset β) (f : β → matrix l m α)
-  (M : matrix m n α) : (∑ a in s, f a) ⬝ M = ∑ a in s, f a ⬝ M :=
-(@finset.sum_hom _ _ _ _ _ s f (λ x, x ⬝ M)
-/- This line does not type-check without `id` and `: _`. Lean did not recognize that two different
-  `add_monoid` instances were def-eq -/
```

Comment:
I figured since this showed up in the diff I may as well sort it out - the problem this refers to doesn't seem to happen any more.